test/e2e/cleanup/iam.go

```diff
@@ -0,0 +1,176 @@
+package cleanup
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/iam/types"
+	"github.com/go-logr/logr"
+
+	"github.com/aws/eks-hybrid/test/e2e/constants"
+	"github.com/aws/eks-hybrid/test/e2e/errors"
+)
+
+type IAMCleaner struct {
+	IAM    *iam.Client
+	Logger logr.Logger
+}
+
+func (c *IAMCleaner) ListRoles(ctx context.Context, filterInput FilterInput) ([]string, error) {
+	var roles []string
+
+	// list-roles does not allow filtering by tags so we have to pull them all
+	// We have the role =* checks to try and limit which roles we bother checking tags for
+	// but we only delete those with the e2e cluster tag
+	paginator := iam.NewListRolesPaginator(c.IAM, &iam.ListRolesInput{})
+
+	for paginator.HasMorePages() {
+		page, err := paginator.NextPage(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("listing IAM roles: %w", err)
+		}
+		for _, role := range page.Roles {
+			if !strings.HasPrefix(*role.RoleName, constants.TestCredentialsStackNamePrefix) {
```

Comment:
should use the TestCredentialsStackNamePrefix in the form of `/prefix/whatever` when creating the roles so we can use the prefix path argument for list roles?